images/Images/test/tTempImage.cc

```diff
@@ -136,9 +136,9 @@ void doIt (TempImage<Int>& scratch)
 // Stream, unstream, and check the image.
 void streamImage (ImageInterface<Int>& img)
 {
-  MemoryIO membuf;
-  CanonicalIO canio (&membuf);
-  AipsIO os (&canio);
+  std::shared_ptr<ByteIO> membuf(new MemoryIO());
+  std::shared_ptr<TypeIO> canio (new CanonicalIO(membuf));
```

Comment:
```suggestion
  std::shared_ptr<MemoryIO> membuf(new MemoryIO());
  std::shared_ptr<CanonicalIO> canio (new CanonicalIO(membuf));
```
Additionally, you could use `std::make_shared` here.